src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -7,394 +7,88 @@
-		&cur_dim_feature_space, &cur_dim_input_space);
-
-}
-
-RandomFourierGaussPreproc::RandomFourierGaussPreproc(
-		const RandomFourierGaussPreproc & feats) :
-	RandomMixin<DensePreprocessor<float64_t>> () {
-
-	randomcoeff_multiplicative=NULL;
-	randomcoeff_additive=NULL;
-
-	SG_ADD(&dim_input_space, "dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&cur_dim_input_space, "cur_dim_input_space",
-			"Dimensionality of the input space.");
-	SG_ADD(&dim_feature_space, "dim_feature_space",
-			"Dimensionality of the feature space.");
-	SG_ADD(&cur_dim_feature_space, "cur_dim_feature_space",
-			"Dimensionality of the feature space.");
-
-	SG_ADD(&kernelwidth, "kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-	SG_ADD(&cur_kernelwidth, "cur_kernelwidth", "Kernel width.", ParameterProperties::HYPER);
-
-	watch_param(
-		"randomcoeff_additive", &randomcoeff_additive,
-		&cur_dim_feature_space);
-
-	watch_param(
-		"randomcoeff_multiplicative", &randomcoeff_multiplicative,
-		&cur_dim_feature_space, &cur_dim_input_space);
-
-	copy(feats);
-}
-
-RandomFourierGaussPreproc::~RandomFourierGaussPreproc() {
-
-	SG_FREE(randomcoeff_multiplicative);
-	SG_FREE(randomcoeff_additive);
-
-}
-
-EFeatureClass RandomFourierGaussPreproc::get_feature_class() {
-	return C_DENSE;
-}
-
-EFeatureType RandomFourierGaussPreproc::get_feature_type() {
-	return F_DREAL;
-}
-
-int32_t RandomFourierGaussPreproc::get_dim_feature_space() const {
-	return ((int32_t) dim_feature_space);
-}
-
-void RandomFourierGaussPreproc::set_dim_feature_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_feature_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_feature_space = dim;
-
-}
-
-int32_t RandomFourierGaussPreproc::get_dim_input_space() const {
-	return ((int32_t) dim_input_space);
-}
-
-void RandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ) {
-	if (kernelwidth2 <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_kernelwidth(const float64_t kernelwidth2 ): kernelwidth2 <= 0 is not allowed");
-	}
-	kernelwidth=kernelwidth2;
-}
-
-float64_t RandomFourierGaussPreproc::get_kernelwidth( ) const {
-	return (kernelwidth);
-}
-
-void RandomFourierGaussPreproc::set_dim_input_space(const int32_t dim) {
-	if (dim <= 0) {
-		throw ShogunException(
-				"void CRandomFourierGaussPreproc::set_dim_input_space(const int32 dim): dim<=0 is not allowed");
-	}
-
-	dim_input_space = dim;
-
-}
-
-bool RandomFourierGaussPreproc::test_rfinited() const {
-
-	if ((dim_feature_space ==  cur_dim_feature_space)
-			&& (dim_input_space > 0) && (dim_feature_space > 0)) {
-		if ((dim_input_space == cur_dim_input_space)&&(Math::abs(kernelwidth-cur_kernelwidth)<1e-5)) {
-
-			// already inited
-			return true;
-		} else {
-			return false;
-		}
-	}
-
-	return false;
+RandomFourierGaussPreproc::RandomFourierGaussPreproc()
+: m_log_width(1.0), m_dim_output(100)
```

Comment:
the effective kernel width should be 1
Also, could you set these values at the declaration